sources/packages/backend/apps/workers/src/services/disbursement-schedule/disbursement-schedule.service.ts

```diff
@@ -53,93 +106,179 @@ export class DisbursementScheduleService extends RecordDataModelService<Disburse
-    const msfaaNumber = { id: msfaaNumberId } as MSFAANumber;
+    return await this.sfasPartTimeApplicationsService.getIndividualPartTimeApplicationByIndividualId(
+      studentId,
+    );
+  }
 
-    // Associate all the disbursements of the given assessment with MSFAA.
-    disbursements.forEach((disbursement) => {
-      disbursement.msfaaNumber = msfaaNumber;
-      disbursement.modifier = systemUser;
-    });
-    await this.repo.save(disbursements);
+  /**
+   * Checks if the MSFAA number is still valid.
+   * If the study period end date of the previously signed MSFAA is less than 2 years
+   * when compared to current study period start date, then MSFAA is considered to be valid.
+   * @param previousSignedDisbursement previously signed disbursement.
+   * @param firstDisbursement first disbursement of the application.
+   * @returns whether the MSFAA Number is valid.
+   */
+  private async isMSFAANumberStillValid(
+    previousSignedDisbursement: DisbursementSchedule,
+    firstDisbursement: DisbursementSchedule,
+  ): Promise<boolean> {
+    const previousStudyEndDate = new Date(
+      previousSignedDisbursement.studentAssessment.offering.studyEndDate,
+    );
+    const currentStudyStartDate = new Date(
+      firstDisbursement.studentAssessment.offering.studyStartDate,
+    );
+    // Previously signed and completed application offering end date in considered the start date.
+    // Start date of the offering of the current application is considered the end date.
+    return this.msfaaNumberService.isMSFAANumberValid(
+      previousStudyEndDate,
+      currentStudyStartDate,
+    );
+  }
+
+  /**
+   * Create and activate the MSFAA number fetched from SFAS signed MSFAA.
+   * @param studentId student Id.
+   * @param offeringIntensity offering intensity.
+   * @param applicationId application id.
+   * @param auditUserId audit user id.
+   * @param sfasSignedMSFAA SFAS MSFAA number and latest SFAS application end date.
+   * @returns created and activated MSFAA number id.
+   */
+  private async createAndActivateSfasMSFAANumber(
+    student: Student,
+    application: Application,
+    offeringIntensity: OfferingIntensity,
+    auditUserId: number,
+    sfasSignedMSFAA: SFASSignedMSFAA,
+  ): Promise<number | null> {
+    // Create and activate new MSFAA number from the SFAS records.
+    const sfasMSFAANumber = new MSFAANumber();
+    sfasMSFAANumber.msfaaNumber = sfasSignedMSFAA.sfasMSFAANumber;
+    sfasMSFAANumber.dateSigned = getISODateOnlyString(
+      sfasSignedMSFAA.latestSFASApplicationEndDate,
+    );
+    sfasMSFAANumber.dateRequested = null;
+    sfasMSFAANumber.serviceProviderReceivedDate = null;
+    sfasMSFAANumber.referenceApplication = application;
+    sfasMSFAANumber.student = student;
+    sfasMSFAANumber.offeringIntensity = offeringIntensity;
+    sfasMSFAANumber.creator = { id: auditUserId } as User;
+    const createdMSFAANumber = await this.msfaaNumberService.createMSFAA(
+      sfasMSFAANumber,
+    );
```

Comment:
I would suggest moving this method to `MSFAANumberSharedService` to be together with the other MSFAA similar methods and not using the `createMSFAA` only to call the `repo.save`.
It can also be renamed to `importMSFAAumberFromSFAS`.